onnxruntime/test/providers/tensorrt/tensorrt_basic_test.cc

```diff
@@ -274,6 +274,85 @@ TEST(TensorrtExecutionProviderTest, MultiThreadsTestWithOneSessionMultiThreadsIn
   RunWithOneSessionMultiThreadsInference(model_name, sess_log_id);
 }
 
+// Test loading same model in different way, when hash id is generated via model name/model content/env metadata
+TEST(TensorrtExecutionProviderTest, TRTMetadefIdGeneratorUsingModelHashing) {
+  auto model_path = ORT_TSTR("testdata/mnist.onnx");
+
+  std::shared_ptr<Model> model;
+  ASSERT_TRUE(Model::Load(model_path, model, nullptr, DefaultLoggingManager().DefaultLogger()).IsOK());
+
+  Graph& graph = model->MainGraph();
+  GraphViewer viewer(graph);
+
+  // get the hash for the model when loaded from file
+  HashValue model_hash;
+  int id = TRTGenerateMetaDefId(viewer, model_hash);
```

Comment:
That's good to test the correctness and functionality of TRTGenerateMetaDefId().

I think it's better to test the TRT EP inference run and check the hash name of the engine cache in the filesystem as well, either manually or in the unit test. So that we can make sure the code path of TRT EP calling TRTGenerateMetaDefId() is okay.

Reply:
This has been manually verified by running FastRCNN locally. Thanks for sharing the instruction!